tests/loggers/test_mosaicml_logger.py

```diff
@@ -194,3 +196,88 @@ def test_auto_add_logger(monkeypatch, platform_env_var, access_token_env_var, lo
     # Otherwise, no logger
     else:
         assert logger_count == 0
+
```

Comment:
for unit tests, instead of testing with a full `trainer.fit()` for each thing, I think we should test 1 full `trainer.fit()` on init and a specific unit and then for each unit, we can test `_get_training_progress_metrics` separately. This way of testing would be more modular and you can much more easily test the training sub progress metric this way. 

You could use magic mock for mocking the state passed to _get_training_progress_metrics:
```
def test_get_training_progress_metrics():
     # test tokens
     logger = MosaicMLLogger()
     state = MagicMock()
     state.max_duration.unit = TimeUnit.TOKEN
     state.timestamp.token.value = ...
     # set other magic mock attributes
     result = logger._get_training_progress_metrics(state)
     # assert stuff

```